src/Map/MapReader.cpp

```diff
@@ -99,6 +109,31 @@ void Map::ReadVersionTag(Stream::Reader& streamReader)
 	}
 }
 
+void Map::ReadSavedGameSecondChunk(Stream::SeekableReader& streamReader)
+{
+	SavedGameDataSection2 savedGameData;
+
+	streamReader.Read(savedGameData.unitCount);
+	streamReader.Read(savedGameData.unknown1);
+	streamReader.Read(savedGameData.unknown2);
+	streamReader.Read(savedGameData.unknown3);
+	streamReader.Read(savedGameData.sizeOfUnit);
+
+	streamReader.Read(savedGameData.objectCount1);
+	streamReader.Read(savedGameData.objectCount2);
+
+	savedGameData.objects1.resize(512 * savedGameData.objectCount1);
+	streamReader.Read(savedGameData.objects1);
+	savedGameData.objects2.resize(savedGameData.objectCount2);
+	streamReader.Read(savedGameData.objects2);
+
+	streamReader.Read(savedGameData.unitID1);
+	streamReader.Read(savedGameData.unitID2);
+
+	//streamReader.Read(savedGameData.unitRecord);
+	streamReader.SeekRelative(0x3BF88);
```

Comment:
This is likely affected by the unit limit patch, which I believe invalidated old saved game files, due to the expansion in size of the unit array.

It would be nice if we could distinguish files for this. Perhaps I should have bumped the file format version number.

Reply:
I would say better late than never on bumping the version, but we might have a lot of angry fans if we bump it for no gain. If you have time, it might be nice to update the documentation file to represent the new expected value.

Reply:
Yeah, I think it's a bit too late to bump the version number for that change. At this point, there will be more new saved game files with the original version number than old saved game files.

It may be possible to distinguish them through other means. Perhaps the field for the number of units, or by using the version tag.

For now, let's just assume the newer version with the expanded unit array and ignore the problem.

I'll update the documentation to add a note. Though I think the SVN is down again, so I won't be able to commit the change yet.